flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/type/BinaryType.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.type;
+
+/**
+ * Binary type, java format is java byte[].
+ */
+public class BinaryType implements AtomicType {
```

Comment:
Why we need BinaryType instead of having a `ArrayType(ByteType)`

Reply:
BinaryType is separate from Array Type in traditional database type system.
But I think we can merge they to let Types.BINARY = new ArrayType(ByteType);